### PR TITLE
"Create Component from Git" doesn't sanitize the suggested component …

### PR DESCRIPTION
**Pull request by vrubezhny**
…name #3294

Fixes: #3294

### CODE REVIEW

File: src/webview/common/sanitize.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *-----------------------------------------------------------------------------------------------*/
+
+/**
+ * Builds a sanitized compponent name from a component folder full path or
+ * a Git-repo URL
+ *
+ * @param pathOrURL an absolute path ot a Git-repo URL
+ * @returns sanitized component name
+ */
+export function buildSanitizedComponentName(pathOrURL?: string): string {
```
Comment (datho7561) [inline src/webview/common/sanitize.ts:13]: I realize this mess of regex is something that I wrote and should have tested more thoroughly. Here is a case that isn't sanitized properly: https://github.com/datho7561/12345--painful--to---manage....git...repo...name

**Review by datho7561 — CHANGES_REQUESTED** [review]
Found an edge case with the name sanitizing function that doesn't work as expected. Otherwise, looks great and works well!

File: src/webview/common/sanitize.ts (hunk 1 above)
Comment (datho7561) [reply src/webview/common/sanitize.ts:13]: It becomes `-painful-to-manage-git-repo-name`, which isn't valid.

Minimal reproducer is `1234-a`:
- `1234` gets reduced to `` in the last step, leaving `-a`. The leading `-` is invalid

Comment (vrubezhny) [reply src/webview/common/sanitize.ts:13]: Yeah, Hope it's fixed now, at least for your examples now it returns  `a` and `painful-to-manage-git-repo-name`

**Review by datho7561 — APPROVED** [review]
Looks good. Thanks, Victor!

**Pull request merged**